internal/k8s/watch.go

```diff
@@ -0,0 +1,47 @@
+package k8s
+
+import (
+	"context"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func (kCli K8sClient) WatchPods(ctx context.Context, ns Namespace, lps []LabelPair) (<-chan *v1.Pod, error) {
+	ch := make(chan *v1.Pod)
+
+	ls := labels.Set{}
+	for _, lp := range lps {
+		ls[lp.Key] = lp.Value
+	}
+
+	watcher, err := kCli.core.Pods(ns.String()).Watch(metav1.ListOptions{LabelSelector: ls.String()})
+	if err != nil {
+		return nil, err
+	}
+
+	go func() {
+		for {
+			select {
+			case event := <-watcher.ResultChan():
```

Comment:
you probably want to use the two argument pull off the channel, so you can tell if the channel is closed by the server, and clean up appropriately.